lib/ansible/plugins/callback/foreman.py

```diff
@@ -80,154 +80,173 @@
+        if option.lower() in ["1", "true", "on"]:
             verify = True
-        elif self.FOREMAN_SSL_VERIFY.lower() in ["0", "false", "off"]:
+        elif option.lower() in ["0", "false", "off"]:
             requests.packages.urllib3.disable_warnings()
-            self._display.warning("SSL verification of %s disabled" %
-                                  self.FOREMAN_URL)
+            self._display.warning(u"SSL verification of %s disabled" %
+                                  self.foreman_url)
             verify = False
         else:  # Set to a CA bundle:
-            verify = self.FOREMAN_SSL_VERIFY
+            verify = option
         return verify
 
-    def send_facts(self, host, data):
+    def send_facts(self):
         """
         Sends facts to Foreman, to be parsed by foreman_ansible fact
         parser.  The default fact importer should import these facts
         properly.
         """
-        data["_type"] = "ansible"
-        data["_timestamp"] = datetime.now().strftime(self.TIME_FORMAT)
-        facts = {"name": host,
-                 "facts": data,
-                 }
-        try:
-            r = requests.post(url=self.FOREMAN_URL + '/api/v2/hosts/facts',
-                              data=json.dumps(facts),
-                              headers=self.FOREMAN_HEADERS,
-                              cert=self.FOREMAN_SSL_CERT,
-                              verify=self.ssl_verify)
-            r.raise_for_status()
-        except requests.exceptions.RequestException as err:
-            print(to_text(err))
-
-    def _build_log(self, data):
-        logs = []
-        for entry in data:
-            source, msg = entry
-            if 'failed' in msg:
-                level = 'err'
-            elif 'changed' in msg and msg['changed']:
-                level = 'notice'
-            else:
-                level = 'info'
-            logs.append({
-                "log": {
-                    'sources': {
-                        'source': source
-                    },
-                    'messages': {
-                        'message': json.dumps(msg)
-                    },
-                    'level': level
-                }
-            })
-        return logs
+        for host, facts in self.facts.items():
+            facts = {
+                "name": host,
+                "facts": {
+                    "ansible_facts": self.facts[host],
+                    "_type": "ansible",
+                    "_timestamp": get_now(),
+                },
+            }
+
+            url = self.foreman_url + '/api/v2/hosts/facts'
+            try:
+                response = self.session.post(url=url, json=facts)
+                response.raise_for_status()
+            except requests.exceptions.RequestException as err:
+                self._display.warning(u'Sending facts to Foreman at {url} failed for {host}: {err}'.format(
+                    host=host, err=to_text(err), url=self.foreman_url))
```

Comment:
Explicit is better than what's happening implicitly:
```suggestion
                    host=to_text(host), err=to_text(err), url=to_text(self.foreman_url)))
```